packages/core/src/utils/BlockchainApiUtil.ts

```diff
@@ -0,0 +1,26 @@
+import { ConfigCtrl } from '../controllers/ConfigCtrl'
+
+// -- Constants -------------------------------------------------------
+export const NAMESPACE = 'eip155'
+// To do remove dev.
+const BLOCKCHAIN_API = 'https://rpc.walletconnect.com'
+
+// -- Utility -------------------------------------------------------
+export const BlockchainApiUtil = {
+  async getIdentity(
+    address: string,
+    chainId: number
+  ): Promise<{ name: string; avatar?: string } | null> {
+    const { projectId } = ConfigCtrl.state
+    const chain_id = `${NAMESPACE}:${chainId}`
+    const endpoint = `${BLOCKCHAIN_API}/v1/identity/${address}?chainId=${chain_id}&projectId=${projectId}`
+
+    const response = await fetch(endpoint)
+    if (response.status === 404) {
```

Comment:
```suggestion
    if (response.status !== 200) {
```

And then we can probably just `throw` to trigger fallback. This way you don't need that `if(profile)` check I commented on above anymore.

Reply:
This is intentional as-is. a 404 is a non-error response saying that the profile was not found. A 500 or 400 would `throw` and trigger the `catch` in the calling function.